### ISSUE TITLE
paramiko 2.10.1 breaks Python 2.7 compatibility

### ISSUE THREAD

**Issue by chipx86** [paramiko/paramiko#2001]
Commit 4c491e299c9b800358b16fa4886d8d94f45abe2e introduces a call to `os.open()` that passes in keyword arguments. This works in Python 3, but generates the following error on Python 2.7:

```
File "/tmp/python-env-cache/lib/python2.7/site-packages/paramiko/rsakey.py", line 161, in write_private_key_file
    password=password,
File "/tmp/python-env-cache/lib/python2.7/site-packages/paramiko/pkey.py", line 570, in _write_private_key_file
    with os.fdopen(os.open(filename, **kwargs), mode="w") as f:
TypeError: open() takes no keyword arguments
```

This can be demonstrated in the Python 2.7 shell:

```python
>>> import os
>>> os.open('abc', **{'flags': 0})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: open() takes no keyword arguments
```

Positional arguments work fine as a (less aesthetically-pleasing, but functional) alternative on 2.7:

```python
>>> import os
>>> os.open('abc', *(0,))
```

**Comment by bitprophet** [comment]
Thanks for the catch; we don't CI or dev on 2 anymore which is why this got overlooked. While we're about to drop it entirely in the next release, this ought to be a quick fix.

**Comment by bitprophet** [comment]
Fixed in 2.10.2 which just went out on PyPI.

**Comment by chipx86** [comment]
Awesome!

Figured that was the case. Made sure to do my homework first on whether Python 2.7 was still being maintained for 2.10 (found the roadmap and the bits about disconnecting from CI — very helpful! Most projects don't have a deprecation plan posted for major Python versions, making it hard to cap dependencies correctly).

Thanks for the fix! Have a good day :)
